pkg/dns/treecache.go

```diff
@@ -166,29 +129,10 @@ func (cache *TreeCache) getValuesForPathWithWildcards(path ...string) []interfac
 	return retval
 }
 
-func (cache *TreeCache) getEntries(recursive bool, path ...string) []interface{} {
-	if cache.rwLock != nil {
-		cache.rwLock.RLock()
-		defer cache.rwLock.RUnlock()
-	}
-	childNode := cache.getSubCache(path...)
-	if childNode == nil {
-		return nil
-	}
-
-	retval := [][]interface{}{{}}
-	childNode.appendValues(recursive, retval)
-	return retval[0]
-}
-
 func (cache *TreeCache) deletePath(path ...string) bool {
```

Comment:
I don't think the comment would make sense here. the treecache is not thread-safe.
It could be used by the same thread/go-routine without locking,
or by multiple threads, using a simple mutex or rwmutex. 
it is up to the caller to choose..